docs: add Faucet API page for programmatic testnet HBAR (#645)
hashgraph/hedera-docs: learn/getting-started/faucet-api.mdx

---
title: "Faucet API"
description: "Fund testnet and previewnet accounts with HBAR programmatically over HTTP, no browser required."
icon: "terminal"
---

> The Hedera Portal faucet exposes an HTTP endpoint so you can fund Hedera accounts programmatically, with no browser, no reCAPTCHA, and no copy-pasting addresses into a web form.

Reviewer: ```suggestion
The Hedera Portal exposes an HTTP faucet endpoint so you can fund testnet and previewnet accounts programmatically from the terminal, scripts, CI/CD pipelines, SDKs, or agentic workflows, with no browser, no reCAPTCHA, and no copy-pasting.
```

> The Faucet API supports funding accounts from the terminal, scripts, CI/CD pipelines, SDKs, or agentic workflows.

Reviewer: ```suggestion
```

</Info>

<Note>
	**Need mainnet HBAR?** This Portal Faucet API serves **testnet and previewnet** only. For mainnet, the ecosystem project HashPort runs a
	community-operated [Faucet API](https://faucet.hashport.network/) that distributes HBAR to onboard new users.
</Note>

---

## Before you start

The Faucet API requires a Hedera Portal account and a Personal Access Token (the web faucet does not):

1. **A Hedera Portal account.** [Sign up](https://portal.hedera.com/register) if you don't have one.
2. **A Personal Access Token (PAT).** Generate one from the Portal UI, following [Create an API Key](/native/tutorials/getting-started/create-api-key).

In your terminal, set your token as an environment variable so the `curl` examples below can read it. The `export` command runs in a macOS or Linux shell (bash or zsh) and lasts for the current terminal session:

```bash
export HEDERA_PAT="<your-personal-access-token>"
```

On Windows PowerShell, use:

```powershell
$env:HEDERA_PAT = "<your-personal-access-token>"
```

---

## Make a request

Send a `POST` to `/api/disbursement/cli` with your PAT as a Bearer token.

> 

Reviewer: ```suggestion

<Info>
	**About the destination:** A Hedera account ID (`0.0.x`) must already exist on the network. An EVM address that has no account yet gets one created
	on the spot through [auto account creation](/learn/core-concepts/accounts/auto-account-creation) when the HBAR lands.
</Info>

```

    "network": "testnet"
  }'
```

### Request fields

| Field        | Required | Description                                                                                 |
| ------------ | -------- | ------------------------------------------------------------------------------------------- |
| `address`    | Yes      | The destination: a Hedera account ID (`0.0.12345`) or an EVM address (`0x…`, 40 hex chars). |
| `amount`     | Yes      | A whole number of HBAR to send, from `1` to `100`.                                          |
| `network`    | No       | `testnet` (default) or `previewnet`.                                                        |
| `sdkVersion` | No       | For tool authors: the SDK version, so usage can be attributed.                              |
| `cliVersion` | No       | For tool authors: the CLI version.                                                          |

<Info>
	**About the destination:** A Hedera account ID (`0.0.x`) must already exist on the network. An EVM address that has no account yet gets one created
	on the spot through [auto account creation](/learn/core-concepts/accounts/auto-account-creation) when the HBAR lands.

> </Info>

Reviewer: ```suggestion
```

Reviewer: i think the callout makes more sense before the curl block
